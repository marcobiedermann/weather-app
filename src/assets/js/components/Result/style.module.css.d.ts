declare const styles: {
  readonly result: string;
};
export = styles;
