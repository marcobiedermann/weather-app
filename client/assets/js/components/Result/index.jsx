import PropTypes from 'prop-types';
import React from 'react';
import { useTranslation } from 'react-i18next';

const Result = (props) => {
  const { addCity, main, name, sys, ...otherProps } = props;
  const { t } = useTranslation();

  return (
    <div {...otherProps}>
      <h3>
        {name}, {sys.country}
      </h3>
      {main.temp}

      <button type="button" onClick={() => addCity(props)}>
        {t('translation:add')}
      </button>
    </div>
  );
};

Result.propTypes = {
  addCity: PropTypes.func,
  main: PropTypes.shape(),
  name: PropTypes.string,
  sys: PropTypes.shape(),
};

Result.defaultProps = {
  addCity: () => {},
  main: null,
  name: '',
  sys: null,
};

export default Result;
