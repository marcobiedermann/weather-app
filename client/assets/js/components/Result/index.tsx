import React, { FC } from 'react';
import { useTranslation } from 'react-i18next';

export interface ResultProps {
  addCity: () => void;
  main: {
    temp: number;
  };
  name: string;
  sys: {
    country: string;
  };
}

const Result: FC<ResultProps> = (props) => {
  const { addCity, main, name, sys } = props;
  const { t } = useTranslation();

  return (
    <div>
      <h3>
        {name}, {sys.country}
      </h3>
      {main.temp}

      <button type="button" onClick={() => addCity(props)}>
        {t('translation:add')}
      </button>
    </div>
  );
};

export default Result;
