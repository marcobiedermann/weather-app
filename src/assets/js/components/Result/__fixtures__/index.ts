const result = {
  clouds: {
    all: 0,
  },
  coord: {
    lat: 52.5244,
    lon: 13.4105,
  },
  dt: 1605434959,
  id: 2950159,
  main: {
    feels_like: 9.39,
    humidity: 76,
    pressure: 1014,
    temp_max: 13.89,
    temp_min: 11,
    temp: 12.61,
  },
  name: 'Berlin',
  rain: null,
  snow: null,
  sys: {
    country: 'DE',
  },
  weather: [
    {
      description: 'clear sky',
      icon: '01d',
      id: 800,
      main: 'Clear',
    },
  ],
  wind: {
    deg: 170,
    speed: 4.1,
  },
};

export default result;
