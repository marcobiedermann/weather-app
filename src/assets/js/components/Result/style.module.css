.result {
  background-color: #929fb3;
  border-radius: 0.5em;
  padding: 1em;
}

.result > :first-child {
  margin: 0;
}

.result > :last-child {
  margin: 0;
}
