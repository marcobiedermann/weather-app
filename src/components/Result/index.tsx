import { PlusCircle } from 'react-feather';
import { useTranslation } from 'react-i18next';
import styles from './style.module.css';

interface ResultProps {
  id: number;
  main: {
    temp: number;
  };
  name: string;
  sys: {
    country: string;
  };
  onClick: (id: number) => void;
}

function Result(props: ResultProps): JSX.Element {
  const { id, main, name, onClick, sys } = props;
  const { t } = useTranslation();

  return (
    <div className={styles.result}>
      <h3>
        {name}, {sys.country}
      </h3>
      {t('intlNumberCelsius', { val: main.temp })}

      <button type="button" onClick={() => onClick(id)}>
        <span style={{ display: 'none' }}>{t('add')}</span>
        <PlusCircle />
      </button>
    </div>
  );
}

export type { ResultProps };
export default Result;
