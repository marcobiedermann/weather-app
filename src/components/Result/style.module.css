.result > :first-child {
  margin: 0;
}

.result > :last-child {
  margin: 0;
}
