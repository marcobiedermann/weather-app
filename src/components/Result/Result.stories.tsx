import type { Meta, StoryObj } from '@storybook/react';
import { fn } from '@storybook/test';
import Result from '.';

const meta: Meta<typeof Result> = {
  component: Result,
  args: {
    id: 2950159,
    name: 'Berlin',
    main: {
      temp: 18.54,
    },
    sys: {
      country: 'DE',
    },
    onClick: fn(),
  },
};

type Story = StoryObj<typeof Result>;

const Default: Story = {};

export { Default };
export default meta;
