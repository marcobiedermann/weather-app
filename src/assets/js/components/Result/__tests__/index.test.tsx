import { render } from '@testing-library/react';
import React from 'react';
import Result from '..';
import resultFixtures from '../__fixtures__';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));

describe('components/Result', () => {
  it('renders correctly', () => {
    expect.assertions(1);

    const { asFragment } = render(<Result {...resultFixtures} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
