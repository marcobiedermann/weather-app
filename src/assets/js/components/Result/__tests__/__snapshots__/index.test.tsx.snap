// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Result renders correctly 1`] = `
<DocumentFragment>
  <div
    class="result"
  >
    <h3>
      Berlin, DE
    </h3>
    12.61
    <button
      type="button"
    >
      translation:add
    </button>
  </div>
</DocumentFragment>
`;
